docs/contributing/frontend.md

```diff
@@ -0,0 +1,133 @@
+  - **mocks** - [Manual mocks](https://jestjs.io/docs/manual-mocks) used by Jest
+  - **@types** - Custom types for dependencies that don't have defined types
+  - **api** - API code as function calls and types
+  - **components** - UI components
+  - **hooks** - Hooks that can be used across the application
+  - **i18n** - Translations
+  - **pages** - Page components
+  - **testHelpers** - Helper functions to help with integration tests
+  - **util** - Helper functions that can be used across the application
+  - **xServices** - XState machines used to fetch data and handle complex scenarios
+- **static** - UI static assets like images, fonts, icons, etc
+
+## Routing
+
+We use [react-router](https://reactrouter.com/en/main) as our routing engine and adding a new route is very easy. If the new route needs to be authenticated, put it under the `<RequireAuth>` route, if it needs to live inside of the dashboard, put it under the `<DashboardLayout>`.
+
+The `RequireAuth` component handles all the authentication logic for the routes and the `DashboardLayout` wraps the route adding a navbar and passing down common dashboard data.
+
+## Pages
+
+Pages are the top-level components of the app. The page component lives under the `src/pages` folder and each page should have its own folder so we can better group the views, tests, utility functions and so on. We use a structure where the page component is responsible to fetch all the data and passing it down to the view. We explain this decision a bit better in the next section.
+
+> ℹ️ Code that is only related to the page should live inside of the page folder but if at some point it is used from other pages or components, you should consider to move it to the `src` level in the `utils`, `hooks` or `components` folder.
+
+### States
+
+A page usually has three states at least: **loading**, **ready** or **success**, and **error** so remember to always handle these scenarios while you are coding a page. We also encourage you to add visual testing for these three states using the `*.stories.ts` file.
+
+## Fetching data
+
+We use [TanStack Query v4](https://tanstack.com/query/v4/docs/react/overview)(previously known as react-query) to fetch data from the API. We also use [XState](https://xstate.js.org/docs/) to handle complex flows with multiple states and transitions.
+
+> ℹ️ We recently changed how we are going to fetch data from the server so you will see a lot of fetches being made using XState machines but feel free to refactor it if you are already touching those files.
+
+### Where to fetch data
+
+Finding the right place to fetch data in React apps is the one million dollar question but we decided to make it only in the page components and pass the props down to the views. This makes it easier to find where data is being loaded and easy to test using Storybook - We are going to talk more about our testing strategy. So you will see components like `UsersPage` and `UsersPageView`.
```

Comment:
> [...] We are going to talk more about our testing strategy. So you will see components like `UsersPage` and `UsersPageView`.

This was a bit confusing for me, talk more about it where? And where will I see the components?

Reply:
Good catch.